packages/patterns/src/components/patterns-manage-button.js

```diff
@@ -18,13 +18,14 @@ import { unlock } from '../lock-unlock';
 function PatternsManageButton( { clientId } ) {
 	const { canRemove, isVisible, managePatternsUrl } = useSelect(
 		( select ) => {
-			const { getBlock, canRemoveBlock, getBlockCount } =
+			const { getBlock, canRemoveBlock, getBlockRootClientId } =
 				select( blockEditorStore );
 			const { canUser } = select( coreStore );
 			const reusableBlock = getBlock( clientId );
+			const rootClientId = getBlockRootClientId( clientId );
 
 			return {
-				canRemove: canRemoveBlock( clientId ),
+				canRemove: canRemoveBlock( clientId, rootClientId ),
```

Comment:
Why does `canRemoveBlock` require a `rootClientID`, while it can derive that itself from the `clientId`? Can't we just fix the function and fix all cases at once?

Commented here too https://github.com/WordPress/gutenberg/pull/62430/files#r1632567240

Reply:
That's a very interesting observation Ella. `rootClientId` seems to be needed for every case.

Reply:
I can't recall; it's probably part of the original API design.

I can follow up on the selector change, but I'm unsure if similar enhancements can be included in release betas/RCs/

Reply:
@Mamaduka It's a bug fix: what happens when you call it with `rootClientId` that is not actually the root? It doesn't make sense to keep it, it eliminates bug across the board for users that  call `canRemoveBlock` without it or wrongly by accident. So it's fine to include.

Reply:
This API was added in [#32457](https://github.com/WordPress/gutenberg/pull/32457/files#diff-aa92c48095c35f17feb834dd919baf52274a73c7b1180ff5acce7b6087b126cfR1314). However, even reading the discussion, there is no mention of why `rootClientID` is required as a parameter, or whether there are cases where `rootClientID` is not required.

Many other selectors accept a `rootClientID` as an option, so this API may have been designed to be consistent with them.